packages/serialized-state-manager/src/internal/client-side-state-manager.ts

```diff
@@ -0,0 +1,14 @@
+export class ClientSideStateManager {
+  private serializedStateByConsumerUid?: Record<string, string | undefined>;
+
+  public setSerializedStates(serializedStates: string): void {
+    this.serializedStateByConsumerUid = JSON.parse(decodeURI(serializedStates));
+  }
+
+  public getSerializedState(consumerUid: string): string | undefined {
+    return (
+      this.serializedStateByConsumerUid &&
```

Comment:
We should probably throw here when this is undefined, since the Integrator forgot to set it.

Reply:
as discussed: won't fix